internal/operands/tekton-pipelines/tekton-pipelines.go

```diff
@@ -163,96 +163,76 @@ func reconcileTektonPipelinesFuncs(pipelines []pipeline.Pipeline) []common.Recon
 func reconcileConfigMapsFuncs(configMaps []v1.ConfigMap) []common.ReconcileFunc {
 	funcs := make([]common.ReconcileFunc, 0, len(configMaps))
 	for i := range configMaps {
-		cm := &configMaps[i]
+		configMap := &configMaps[i]
 		funcs = append(funcs, func(request *common.Request) (common.ReconcileResult, error) {
-			namespace := request.Instance.Spec.Pipelines.Namespace
-			cm.Namespace = namespace
+			if request.Instance.Spec.TektonPipelines != nil && request.Instance.Spec.TektonPipelines.Namespace != "" {
+				configMap.Namespace = request.Instance.Spec.TektonPipelines.Namespace
+			}
 			return common.CreateOrUpdate(request).
-				ClusterResource(cm).
+				ClusterResource(configMap).
 				WithAppLabels(operandName, operandComponent).
-				UpdateFunc(func(newRes, foundRes client.Object) {
-					newCM := newRes.(*v1.ConfigMap)
-					foundCM := foundRes.(*v1.ConfigMap)
-					foundCM.Data = newCM.Data
-				}).
 				Reconcile()
 		})
 	}
 	return funcs
 }
 
-func reconcileServiceAccountsFuncs(r *common.Request, sas []v1.ServiceAccount) []common.ReconcileFunc {
-	funcs := make([]common.ReconcileFunc, 0, len(sas))
-	for i := range sas {
-		sa := &sas[i]
-		// deploy pipeline SA only in `^(openshift|kube)-` namespaces
-		if !namespaceRegex.MatchString(r.Instance.Spec.Pipelines.Namespace) && sa.Name == "pipeline" {
-			continue
+func reconcileServiceAccountsFuncs(r *common.Request, serviceAccounts []v1.ServiceAccount) []common.ReconcileFunc {
+	funcs := make([]common.ReconcileFunc, 0, len(serviceAccounts))
+	for i := range serviceAccounts {
+		serviceAccount := &serviceAccounts[i]
+
+		if r.Instance.Spec.TektonPipelines != nil && r.Instance.Spec.TektonPipelines.Namespace != "" {
+			// deploy pipeline SA only in `^(openshift|kube)-` namespaces
+			if !namespaceRegex.MatchString(r.Instance.Spec.TektonPipelines.Namespace) && serviceAccount.Name == "pipeline" {
+				continue
+			}
 		}
 
 		funcs = append(funcs, func(request *common.Request) (common.ReconcileResult, error) {
 			namespace := request.Instance.Namespace
-			if sa.Namespace == "" {
-				sa.Namespace = namespace
+			if serviceAccount.Namespace == "" {
```

Comment:
IMHO, it's not correct to use the namespace stored in the bundle. I think we should overwrite the namespace with the one specified in `.Spec.TektonPipelines.Namespace`.

@ksimon1 What do you think?

Reply:
all objects related to pipelines (service accounts, role, role binding) should be deployed to the same namespace as pipelines (in this case value from .Spec.TektonPipelines.Namespace - if possible)

Reply:
Okay, so instead of:
```
namespace := request.Instance.Namespace
if serviceAccount.Namespace == "" {
	serviceAccount.Namespace = namespace
}
```
We'll do:
```
if request.Instance.Spec.TektonPipelines != nil && request.Instance.Spec.TektonPipelines.Namespace != "" {
	serviceAccount.Namespace = request.Instance.Spec.TektonPipelines.Namespace
}
```
In all resources?

Reply:
yes, but for Tekton Tasks, it will be the other field: `request.Instance.Spec.TektonTasks.Namespace`.

And when the namespace in `request.Instance.Spec.TektonPipelines.Namespace` is `""`, then the namepsace on resource needs to be set to the `request.Namespace`.

Reply:
Yes, of course, in Tekton Tasks operand it's different. So then:
```
serviceAccount.namespace := request.Instance.Namespace
if request.Instance.Spec.TektonPipelines != nil && request.Instance.Spec.TektonPipelines.Namespace != "" {
	serviceAccount.Namespace = request.Instance.Spec.TektonPipelines.Namespace
}
```

Reply:
yes and the same should be done for all objects (except ClusterRole)

Reply:
Done.